base/util_testing.go

```diff
@@ -568,23 +568,23 @@ func SetUpBenchmarkLogging(logLevel LogLevel, logKeys LogKey) (teardownFn func()
 	}
 }
 
-func setTestLogging(logLevel LogLevel, logKeys LogKey, caller string) (teardownFn func()) {
+func setTestLogging(logLevel LogLevel, caller string, logKeys ...LogKey) (teardownFn func()) {
 	initialLogLevel := LevelInfo
-	initialLogKey := KeyHTTP
+	initialLogKey := logKeyMask(KeyHTTP)
 
 	// Check that a previous invocation has not forgotten to call teardownFn
 	if *consoleLogger.LogLevel != initialLogLevel ||
-		*consoleLogger.LogKey != initialLogKey {
+		*consoleLogger.LogKeyMask != *initialLogKey {
 		panic("Logging is in an unexpected state! Did a previous test forget to call the teardownFn of SetUpTestLogging?")
 	}
 
 	consoleLogger.LogLevel.Set(logLevel)
-	consoleLogger.LogKey.Set(logKeys)
+	consoleLogger.LogKeyMask = logKeyMask(logKeys...)
```

Comment:
Looks like we need to still do an atomic set of the full key mask value here, rather than changing the pointer to prevent races.

```
2020-01-13T11:50:33.199Z [INF] base.TestSetUpTestLogging: Setup logging: level: debug - keys: [DCP Sync]
==================
WARNING: DATA RACE
Write at 0x00c000176828 by goroutine 38:
  github.com/couchbase/sync_gateway/base.setTestLogging()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/util_testing.go:582 +0x16f
  github.com/couchbase/sync_gateway/base.SetUpTestLogging()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/util_testing.go:543 +0x1d5
  github.com/couchbase/sync_gateway/base.TestSetUpTestLogging()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/util_test.go:575 +0x266
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199

Previous read at 0x00c000176828 by goroutine 82:
  github.com/couchbase/sync_gateway/base.(*ConsoleLogger).shouldLog()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/logger_console.go:109 +0xa5
  github.com/couchbase/sync_gateway/base.logTo()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/logging.go:461 +0xad
  github.com/couchbase/sync_gateway/base.(*Replicator).runContinuousReplication.func2()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/logging.go:426 +0x238

Goroutine 38 (running) created at:
  testing.(*T).Run()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:960 +0x651
  testing.runTests.func1()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:1202 +0xa6
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
  testing.runTests()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:1200 +0x521
  testing.(*M).Run()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:1117 +0x2ff
  main.main()
      _testmain.go:380 +0x223

Goroutine 82 (finished) created at:
  github.com/couchbase/sync_gateway/base.(*Replicator).runContinuousReplication()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/replicator.go:222 +0x508
  github.com/couchbase/sync_gateway/base.(*Replicator).startReplication()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/replicator.go:112 +0x1e6
  github.com/couchbase/sync_gateway/base.(*Replicator).Replicate()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/replicator.go:49 +0x16f
  github.com/couchbase/sync_gateway/base.TestReplicator()
      /Users/benbrooks/dev/sync_gateway/master/godeps/src/github.com/couchbase/sync_gateway/base/replicator_test.go:20 +0x23f
  testing.tRunner()
      /usr/local/Cellar/go/1.13.4/libexec/src/testing/testing.go:909 +0x199
==================

```

Reply:
I think this was an oversight - logKeyMask.Set() should have been switched to take a LogKeyMask instead of LogKey.  Have updated.